File: src/test/resources/control/control.js
/**
 *
 * @param a {number}
 * @param b {boolean}
 * @return {number}
 */
function control(a, b) {
    if (x) {
        if (y) doY()
    } else if (z) doZ


    if (b) {
        do {
            a += 1
        } while (!b);
        while (a < 10) {
            a += 2
        }
    } else {
        if (!b) a = 100;
        for (var i=0; i < 3; i++) {
            a += 10
        }
    }
    return a;
}

function f() {
    control(10, true);
    control(15, false);
}

File: src/test/resources/control/control.scala
def control(a: Double, b: Boolean) = {
  var aa = a
  if (b) {
    do aa += 1
    while (!b)
    {
      while (!b) {
        aa += 2
        true
      }
    }
  }
  else {
    if (!b) aa = 100

    {
      var i = 0
      while (i < 3) {
        aa += 10
        i += 1
      }
    }
  }
  return aa
}
def f() = {
  control(10, true)
  control(15, false)
}
